### PR TITLE
fix basic auth

### PR DESCRIPTION
**Pull request by kunmingg**
Resolves https://github.com/kubeflow/kubeflow/issues/3562

Most changes are from https://github.com/kubeflow/kubeflow/pull/3404

Related: https://github.com/kubeflow/kubeflow/pull/3631

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/manifests/204)

### CODE REVIEW

**Comment by kunmingg** [comment]
/assign @jlewi 
/assign @lluunn 
Ready for review

File: common/ambassador/base/deployment.yaml [hunk 1]
```diff
@@ -2,6 +2,7 @@ apiVersion: apps/v1beta1
 kind: Deployment
 metadata:
   name: ambassador
+  namespace: istio-system
```
Comment (lluunn) [inline common/ambassador/base/deployment.yaml]: If we didn't specify namespace, how did it end up in namespace kubeflow?
I mean, shouldn't there be a way to set namespace for all components

Comment (kunmingg) [reply common/ambassador/base/deployment.yaml]: Currently kfctl code use kubeflow ns by default

**Pull request merged**